src/android/com/megster/cordova/BluetoothSerial.java

```diff
@@ -608,13 +677,24 @@ private void executeQueueHandler() {
     }
 
     private synchronized void connectClassic(BluetoothDevice device, boolean secure, CallbackContext callbackContext) {
-        executeQueueHandler();
+
         Log.d(TAG, "## Connect to classic " + device.getName());
         if (device != null) {
 
             connectCallback = callbackContext;
-            bluetoothSerialService.start(device);
-            bluetoothSerialService.connect(device, secure);
+            //Based on device, we need to either call Start or Connect
+            if(SPPProfileDevice.isSupported(device)) { //For devices which need SPP Listener, we only start Accept thread, call connect only for pairing purpose
```

Comment:
For devices which need SPP Listener, we only start Accept thread, call connect only for pairing purpose